src/ToolBox/SOS/Strike/util.cpp

```diff
@@ -3344,20 +3361,41 @@ void DumpMDInfoFromMethodDescData(DacpMethodDescData * pMethodDescData, DacpReJi
 
     if (!fStackTraceFormat)
     {
-        ExtOut("Method Name:  %S\n", wszNameBuffer);
+        ExtOut("Method Name:          %S\n", wszNameBuffer);
 
         DacpMethodTableData mtdata;
         if (SUCCEEDED(mtdata.Request(g_sos, pMethodDescData->MethodTablePtr)))
         {
-            DMLOut("Class:        %s\n", DMLClass(mtdata.Class));
+            DMLOut("Class:                %s\n", DMLClass(mtdata.Class));
         }            
 
-        DMLOut("MethodTable:  %s\n", DMLMethodTable(pMethodDescData->MethodTablePtr));
-        ExtOut("mdToken:      %p\n", SOS_PTR(pMethodDescData->MDToken));
-        DMLOut("Module:       %s\n", DMLModule(pMethodDescData->ModulePtr));
-        ExtOut("IsJitted:     %s\n", pMethodDescData->bHasNativeCode ? "yes" : "no");
-        DMLOut("CodeAddr:     %s\n", DMLIP(pMethodDescData->NativeCodeAddr));                
+        DMLOut("MethodTable:          %s\n", DMLMethodTable(pMethodDescData->MethodTablePtr));
+        ExtOut("mdToken:              %p\n", SOS_PTR(pMethodDescData->MDToken));
+        DMLOut("Module:               %s\n", DMLModule(pMethodDescData->ModulePtr));
+        ExtOut("IsJitted:             %s\n", pMethodDescData->bHasNativeCode ? "yes" : "no");
 
+        DMLOut("Current CodeAddr:     %s\n", DMLIP(pMethodDescData->NativeCodeAddr));                
+
+        CLRDATA_ADDRESS codeAddrs[kcMaxTieredVersions];
+        int cCodeAddrs;
+
+        ReleaseHolder<ISOSDacInterface5> sos5;
+        if (SUCCEEDED(g_sos->QueryInterface(__uuidof(ISOSDacInterface5), &sos5)) && 
+            SUCCEEDED(sos5->GetTieredVersions(pMethodDescData->MethodDescPtr, 
+                                                                (int)pMethodDescData->rejitDataCurrent.rejitID,
+                                                                codeAddrs,
+                                                                kcMaxTieredVersions,
+                                                                &cCodeAddrs)))
+        {
+            // Want to print out the previous ones, but the array contains all of the native code addresses,
+            // including the active one (which will be last). So start at one before the last one and work
+            // our way down      
+            for(int i = cCodeAddrs - 2; i >= 0; --i)
+            {
+                DMLOut("Previous CodeAddr:    %s\n", DMLIP(codeAddrs[i]));
```

Comment:
It would be helpful to get some info about the code versions that correspond to these code addresses, for example which tier is it? In the case there are no previous addresses it could still be useful to know whether the current code address represents the Tier0 code or if tiering is disabled for that method and it represents the permanent code.